File: testPrograms/ts.cc
#include <iostream>

using namespace std;

#include <stdlib.h>

#include "pluton/service.h"


int
main(int argc, char** argv)
{
  pluton::service S("ts");

  if (!S.initialize()) {
    cerr << S.getFault().getMessage("ts") << endl;
    exit(1);
  }

  while (S.getRequest()) {
    string req;
    string con;
    string key;
    cout << "SK = " << S.getServiceKey(key);
    S.getRequestData(req);
    S.getContext("context", con);
    cerr << "Mas Request: " << req << " C=" << con << endl;
    string sa;
    string sf;
    S.getServiceApplication(sa);
    S.getServiceFunction(sf);

    S.sendFault(1, "ERROR");
    S.sendResponse(sf);
  }

  if (S.hasFault()) cerr << "Error: " << S.getFault().getMessage() << endl;

  return 0;
}

File: wrappers/perl/ts.pl
#! /usr/local/bin/perl -w

use Yahoo::pluton;

print "Client API: ", Yahoo::pluton::client::getAPIVersion(), "\n";
print "Service APU: ", Yahoo::pluton::service::getAPIVersion(), "\n";

my $s = Yahoo::pluton::service->new("perlbaby");
$s->initialize();

while ($s->getRequest()) {
    my $str = $s->getRequestData();
#    print STDERR "Have req of $str\n";
    if ($s->hasFault()) {
	print STDERR "getRequestData Fault: ", $s->getFault()->getMessage("getRequest"), "\n";
	exit(0);
    }

    if (length($str) == 0) {
	$s->sendFault(1, "Empty request is invalid");
    }
    else {
	my $response = "Your request is " . length($str) . " bytes long";
	$s->sendResponse($response);
    }
}

if ($s->hasFault()) {
    print STDERR "Service Fault: ", $s->getFault()->getMessage("getRequest"), "\n";
    exit(0);
}

$s->terminate();
exit(0);

